implement-shell-tools/wc/wc.js

```diff
@@ -0,0 +1,71 @@
+const { promises: fs } = require('fs');
+const path = require('path');
+const { program } = require('commander');
+
+program
+  .description('Count lines, words, and characters in the specified files')
+  .option('-l, --lines', 'Count the lines')
+  .option('-w, --words', 'Count the words')
+  .option('-c, --characters', 'Count the characters')
+  .parse(process.argv);
+
+const options = program.opts();
+const files = program.args;
+
+if (files.length === 0) {
+  console.error("Error: No files provided.");
+  process.exit(1);
+}
+
+async function countFileStats(filePath, options) {
+  try {
+    const data = await fs.readFile(filePath, 'utf8');
+    
+    const lines = data.split('\n').length;
+    const words = data.split(/\s+/).filter(Boolean).length;
```

Comment:
This section of the code is processing `data` twice: Once to split it into lines, and another one to split it into words. While this is correct, in very large files, this could potentially be quite slow. 

Is there a way we could avoid processing the entirety of `data` twice?

This is a stretch goal, as the current program is already correct.